src/main/java/org/mockito/internal/session/DefaultMockitoSessionBuilder.java

```diff
@@ -0,0 +1,33 @@
+package org.mockito.internal.session;
+
+import org.mockito.MockitoSession;
+import org.mockito.internal.framework.DefaultMockitoSession;
+import org.mockito.internal.util.ConsoleMockitoLogger;
+import org.mockito.quality.Strictness;
+import org.mockito.session.MockitoSessionBuilder;
+
+public class DefaultMockitoSessionBuilder implements MockitoSessionBuilder {
+
+    private Object testClassInstance;
+    private Strictness strictness;
+
+    @Override
+    public MockitoSessionBuilder initMocks(Object testClassInstance) {
+        this.testClassInstance = testClassInstance;
+        return this;
+    }
+
+    @Override
+    public MockitoSessionBuilder strictness(Strictness strictness) {
+        this.strictness = strictness;
+        return this;
+    }
+
+    @Override
+    public MockitoSession startMocking() {
+        //Configure default values
+        Object effectiveTest = this.testClassInstance == null ? new Object() : this.testClassInstance;
```

Comment:
This is confusing to me `Mockito.session(this)`, especially since the instance could be `null`, if one wants it that way. And this is a builder.

I like this part of the API the way it is. The listener could be extracted in a private method ?